consensus/spos/bls/subroundBlock.go

```diff
@@ -265,22 +265,26 @@ func (sr *subroundBlock) sendBlockHeader(headerHandler data.HeaderHandler, marsh
 }
 
 func (sr *subroundBlock) createHeader() (data.HeaderHandler, error) {
-	hdr := sr.BlockProcessor().CreateNewHeader(uint64(sr.Rounder().Index()))
+	var nonce uint64
+	var prevHash []byte
+	var prevRandSeed []byte
 
 	currentHeader := sr.Blockchain().GetCurrentBlockHeader()
-	var prevRandSeed []byte
 	if check.IfNil(currentHeader) {
-		hdr.SetNonce(1)
-		hdr.SetPrevHash(sr.Blockchain().GetGenesisHeaderHash())
-
+		nonce = 1
+		prevHash = sr.Blockchain().GetGenesisHeaderHash()
 		prevRandSeed = sr.Blockchain().GetGenesisHeader().GetRandSeed()
 	} else {
-		hdr.SetNonce(currentHeader.GetNonce() + 1)
-		hdr.SetPrevHash(sr.Blockchain().GetCurrentBlockHeaderHash())
-
+		nonce = currentHeader.GetNonce() + 1
+		prevHash = sr.Blockchain().GetCurrentBlockHeaderHash()
 		prevRandSeed = currentHeader.GetRandSeed()
 	}
 
+	round := uint64(sr.Rounder().Index())
+	hdr := sr.BlockProcessor().CreateNewHeader(round, nonce)
```

Comment:
i think there is no more need to set round and nonce outside of createNewHeader -> at creation the blockprocessor could save the round and nonce directly in the created header.

Reply:
done